sdks/python/apache_beam/io/gcsio.py

```diff
@@ -555,11 +596,15 @@ def _get_segment(self, start, size):
     """Get the given segment of the current GCS file."""
     if size == 0:
       return ''
+    # The objects self.downloader and self.download_stream may be recreated if
```

Comment:
This is fine because we will only recreate these objects after the timeout, which is after 60 seconds. The thread will have started well before then and saved these members.